great_expectations/rule_based_profiler/parameter_builder/multi_batch_parameter_builder.py

```diff
@@ -0,0 +1,131 @@
+import uuid
+from abc import abstractmethod
+from typing import Dict, List, Optional, Union
+
+import great_expectations.exceptions as ge_exceptions
+from great_expectations import DataContext
+from great_expectations.core.batch import Batch, BatchRequest
+from great_expectations.rule_based_profiler.domain_builder.domain import Domain
+from great_expectations.rule_based_profiler.parameter_builder.parameter_builder import (
+    ParameterBuilder,
+)
+from great_expectations.rule_based_profiler.parameter_builder.parameter_container import (
+    ParameterContainer,
+)
+from great_expectations.rule_based_profiler.util import (
+    get_parameter_value_and_validate_return_type,
+)
+from great_expectations.validator.validator import Validator
+
+
+class MultiBatchParameterBuilder(ParameterBuilder):
+    """
+    Defines the abstract MultiBatchParameterBuilder class
+
+    MultiBatchParameterBuilder checks that there are multiple batch ids passed to its "_build_parameters()" method,
+    and uses a configured batch_request parameter to obtain them if they are not.
+
+    This is an abstract class in the sense that instead of implementing the interface method, "_build_parameters()",
+    this method will remain abstract (for subclasses to implement) and the present class will contain useful utilities
+    for the subclasses.  For example, getting the effective list of batch IDs (whether or not it was passed as argument
```

Comment:
I think this line is a little confusing when compared with the `__init__` definition. Do we want to add `batch_ids` as an optional parameter in the `__init__`? If not, I think we should change this example.

Reply:
@talagluck The comment was confusing.  It will be changed in a "cleanup" PR.